### PR TITLE
vendor: update CI dependencies with latest releases

### PR DESCRIPTION
**Pull request by aanm**
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/7102)

### CODE REVIEW

**Comment by aanm** [comment]
test-me-please https://github.com/cilium/cilium/issues/7105

File: Gopkg.toml [hunk 1]
```diff
@@ -289,14 +289,14 @@ required = [
 
 [[constraint]]
   name = "github.com/onsi/ginkgo"
-  revision = "ba8e856bb854d6771a72ddf6497a42dad3a0c971"
+  revision = "v1.7.0"
```
Comment (eloycoto) [inline Gopkg.toml:292]: I'm not sure if this is good TBH. 1.7 gingko added the JustAfterEach function and I'm not sure that works as we want/expect.

Comment (aanm) [reply Gopkg.toml:292]: It seems that everything is working as expected

**Comment by aanm** [comment]
test-me-please
```
18:28:18     k8s1-1.8: 	-t "cilium/cilium:"latest"" .
18:28:34     k8s1-1.8: error checking context: 'file ('/home/vagrant/go/src/github.com/cilium/cilium/bpf/bpf_lb.ll-b11aa48a') not found or excluded by .dockerignore'.
18:28:34     k8s1-1.8: Makefile:163: recipe for target 'docker-image-no-clean' failed
18:28:34     k8s1-1.8: make: *** [docker-image-no-clean] Error 1
18:28:34 The SSH command responded with a non-zero exit status. Vagrant
```

File: Gopkg.toml (hunk 1 above)
Comment (ianvernon) [reply Gopkg.toml:292]: @eloycoto what would make you comfortable with using `v1.7.0`? Should failure cases in the Ginkgo tests be induced via adding something like `Expect(false).Should(BeTrue())` and see how `JustAfterEach` behaves in our CI?

**Review by ianvernon — APPROVED** [review]

**Comment by aanm** [comment]
test-me-please

```
Required context class hudson.FilePath is missing
Perhaps you forgot to surround the code with a step that provides this, such as: node
```

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

File: Gopkg.toml (hunk 1 above)
Comment (eloycoto) [reply Gopkg.toml:292]: I think so, if something fails in the JustAfterEach the Fail will work(as normal) something that we don't want, due is not handle by ginkgo-ext, but I didn't test with 1.7 so I'm not sure about this. Just a concern here.

**Comment by borkmann** [comment]
(As discussed offline with Eloy, we'll revert worst case.)

**Pull request merged**